projects/sbb-esta/angular-business/src/lib/header/navbutton/navbutton.component.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+
+@Component({
+  selector: 'sbb-navbutton',
+  templateUrl: './navbutton.component.html',
+  styleUrls: ['./navbutton.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NavbuttonComponent implements AfterViewInit, OnDestroy {
+  /**
+   * Css class of a sbb-navbutton
+   */
+  @HostBinding('class.sbb-navbutton') cssClass = true;
+
+  /**
+   * Button (or equivalent) component that navbutton wraps around.
+   */
+  @ViewChild('button', { static: true }) childButton: ElementRef;
+
+  /**
+   * Observer to detect when an eventual dropdown attached to childButton is expanded or collapsed.
+   * Used to propagate change detection.
+   */
+  private _attributeObserver: MutationObserver;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+
+  ngAfterViewInit() {
+    this._checkChild();
+    this._attributeObserver = new MutationObserver(() => this._changeDetectorRef.markForCheck());
```

Comment:
Prefer usage of https://material.angular.io/cdk/observers/api, since this already considers edge cases

Reply:
Very interesting, didn't know about these, looks like they'll help handling those cases even better.